app/src/main/java/com/fstyle/structure_android/utils/validator/Validator.java

```diff
@@ -155,95 +146,50 @@ private boolean validate(Object factor, Rule[] rules, boolean isOptional) {
-                    if (!valid) {
-                        isValid = false;
-                    }
-                }
-            } catch (IllegalAccessException e) {
-                Log.e(TAG, "validate: ", e);
-            }
-            try {
-                if (model != null) {
-                    mModelCache = model.clone();
-                }
-                mViewCache = view;
-            } catch (CloneNotSupportedException e) {
-                mModelCache = null;
-                mViewCache = null;
-                Log.e(TAG, "validate: ", e);
+            Object obj = field.get(object);
+            boolean valid = validate(obj, rules, isOptional);
+            if (!valid) {
+                isValid = false;
             }
         }
         return isValid;
     }
 
-    public <T extends BaseModel> boolean validateAll(T model, boolean onlyValidateChange) {
-        return validateAll(model, null, onlyValidateChange);
-    }
-
-    public <V extends BaseViewModel> boolean validateAll(V view, boolean onlyValidateChange) {
-        return validateAll(null, view, onlyValidateChange);
-    }
-
-    public <T extends BaseModel> Validator prepare(T model) {
-        try {
-            mModelCache = model.clone();
-        } catch (CloneNotSupportedException e) {
-            Log.e(TAG, "prepare: ", e);
-        }
-        return this;
-    }
-
-    @StringRes
-    public int getErrorByTypeInt(@ValidType int type) {
-        int index = mAllErrorMessage.indexOfValue(type);
-        if (index == -1) {
-            return R.string.empty;
-        }
-        return mAllErrorMessage.keyAt(index);
-    }
-
-    public String getErrorByType(@ValidType int type) {
-        int index = mAllErrorMessage.indexOfValue(type);
-        if (index == -1 || mAllErrorMessage.keyAt(index) == -1) {
-            return null;
-        }
-        return mContext.getString(mAllErrorMessage.keyAt(index));
-    }
-
     public Subscription initNGWordPattern() {
-        return Observable.create((Observable.OnSubscribe<Pattern>) subscriber -> {
-            try {
-                BufferedReader reader = new BufferedReader(
-                        new InputStreamReader(mContext.getAssets().open("ng-word")));
-                StringBuilder buffer = new StringBuilder();
-                String line;
-                while ((line = reader.readLine()) != null) {
-                    if (buffer.length() == 0) {
-                        buffer.append("(").append(line.toLowerCase(Locale.ENGLISH));
-                    } else {
-                        buffer.append("|").append(line.toLowerCase(Locale.ENGLISH));
+        return Observable.create(new Observable.OnSubscribe<Pattern>() {
+            @Override
+            public void call(Subscriber<? super Pattern> subscriber) {
+                try {
+                    BufferedReader reader = new BufferedReader(
+                            new InputStreamReader(mContext.getAssets().open("ng-word")));
```

Comment:
ERROR: [I18N][DM_DEFAULT_ENCODING] Found reliance on default encoding in com.fstyle.structure_android.utils.validator.Validator$3.call(Subscriber): new java.io.InputStreamReader(InputStream)
http://findbugs.sourceforge.net/bugDescriptions.html#DM_DEFAULT_ENCODING